scripts/extend_index_yaml_test.py

```diff
@@ -0,0 +1,270 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/extend_index_yaml.py."""
+
+from __future__ import annotations
+
+from unittest import mock
+
+from core.tests import test_utils
+
+from . import extend_index_yaml
+
+
+class ExtendIndexYamlTest(test_utils.GenericTestBase):
+    """Class for testing the extend_index_yaml script."""
+
+    def test_extend_index_yaml_with_changes(self):
+        index_yaml = """indexes:
+
+- kind: AppFeedbackReportModel
+  properties:
+  - name: created_on
+  - name: scrubbed_by
+
+- kind: BlogPostRightsModel
+  properties:
+  - name: blog_post_is_published
+  - name: editor_ids
+  - name: last_updated
+    direction: desc
+"""
+        web_inf_index_yaml = """indexes:
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: status
+  - name: next_scheduled_check_time
+"""
+        expected_index_yaml = """indexes:
+
+- kind: AppFeedbackReportModel
+  properties:
+  - name: created_on
+  - name: scrubbed_by
+
+- kind: BlogPostRightsModel
+  properties:
+  - name: blog_post_is_published
+  - name: editor_ids
+  - name: last_updated
+    direction: desc
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: status
+  - name: next_scheduled_check_time
+"""
+
+        with mock.patch('builtins.open', mock.mock_open()) as mock_open_file:
```

Comment:
Please use mocking that is available in `GenericTestBase`. Ditto elsewhere.

Reply:
Done